### PR TITLE
Upgrade symfony to ^3.0

### LINKED ISSUE #2106: Symfony 3 Compatibility
| Q | A |
| --- | --- |
| Bug? | no |
| Version Used | n/a |
| New Feature? | yes |

This is a meta issue to track progress of all the PRs here and in other packages for Symfony 3 compatibility.
#### Packages to update
- [x] `doctrine/doctrine-bundle: ^1.5.1`
- [x] `doctrine/doctrine-fixtures-bundle: ^2.3`
- [x] `doctrine/phpcr-bundle: ^1.3` -> #2013, #2125 
- [x] `friendsofsymfony/http-cache: t.b.d.`
- [x] `gedmo/doctrine-extensions: t.b.d.`
- [x] `guzzle/guzzle` -> `guzzlehttp/guzzle: ^6.0` (no symfony deps anymore) -> https://github.com/sulu/sulu/pull/3070
- [x] `liip/theme-bundle: ^1.4` -> https://github.com/sulu/SuluThemeBundle/pull/5
- [x] `massive/build-bundle: t.b.d.` -> https://github.com/massiveart/MassiveBuildBundle/pull/12 https://github.com/massiveart/MassiveBuildBundle/pull/13
- [x] `massive/search-bundle: t.b.d.`
- [x] `matthiasnoback/symfony-dependency-injection-test: ^0.7.6` -> #2105 
  - [x] `matthiasnoback/symfony-config-test: ^1.3.1` -> #2105 
- [x] `phpcr/phpcr-shell: t.b.d.`
- [x] `pulse00/ffmpeg-bundle: ^0.6` -> #2114 
- [x] `stof/doctrine-extensions-bundle: ^1.2.2`
- [x] `sulu/document-manager: t.b.d.` -> https://github.com/sulu/sulu-document-manager/pull/103
  - [x] `symfony-cmf/core-bundle: t.b.d.` -> removed this dependency
- [ ] `symfony-cmf/routing: ^2.0` (will be released end of 2016)
- [ ] `symfony-cmf/routing-bundle: 2.0` (will be released end of 2016)
- [x] `symfony/monolog-bundle: ^2.8.2`
- [x] `willdurand/hateoas-bundle: ^1.1` -> https://github.com/sulu/sulu/pull/3071
  - [x] `willdurand/hateoas: ^2.9`
  - [x] `jms/serializer-bundle: ^1.1`
#### Packages which need a compatibility PR itself
- [x] `friendsofsymfony/http-cache`: Since `^1.4.1` it's compatible to Symfony 3 on it's own, but still requiring `guzzle/guzzle: ~3.8`
- [x] `gedmo/doctrine-extensions` (only require-dev dependency)
- [x] `massive/build-bundle` ->  massiveart/MassiveBuildBundle#12 massiveart/MassiveBuildBundle#13
- [x] `massive/search-bundle` -> massiveart/MassiveSearchBundle#83
- [x] `phpcr/phpcr-shell`
- [x] `dantleech/phpcr-migrations-bundle` -> dantleech/phpcr-migrations-bundle#8
- [x] `sulu/document-manager` -> https://github.com/sulu/sulu-document-manager/pull/103
  - [x] `symfony-cmf/core-bundle` -> symfony-cmf/CoreBundle#181
- [ ] `symfony-cmf/routing-bundle` -> symfony-cmf/RoutingBundle#333
  - [x] sonata-project/SonataDoctrinePhpcrAdminBundle#363
#### Sulu bundles without deprecation notices
- [ ] `AdminBundle`
- [ ] `CategoryBundle`
- [ ] `CollaborationBundle`
- [ ] `ContactBundle`
- [ ] `ContentBundle`
- [x] `CoreBundle` -> #2743 
- [x] `DocumentManagerBundle` -> #2720 
- [x] `GeneratorBundle` -> #2721
- [ ] `HashBundle`
- [x] `HttpCacheBundle` -> #2722 
- [x] `LocationBundle`
- [x] `MarkupBundle` -> #2723
- [ ] `MediaBundle`
- [x] `PersistenceBundle` -> #2725
- [x] `PreviewBundle` -> #2726
- [ ] `ResourceBundle`
- [x] `RouteBundle` -> #2727
- [ ] `SearchBundle`
- [ ] `SecurityBundle`
- [ ] `SnippetBundle`
- [ ] `TagBundle`
- [ ] `TestBundle`
- [x] `TranslateBundle` -> #2737 
- [ ] `WebsiteBundle`
- [x] `WebsocketBundle` -> #2728
#### Other steps
- [x] Modify test runner to fail on bundles which have already no deprecation notices, to prevent the introduction of new ones. -> #2120, #2369, #2745
- [x] Add Symfony 3 to Travis matrix with allow failures (if deps are met and installable).
- [ ] Fix possible remaining errors and arm Symfony 3 tests.
- [ ] Do the full update to Symfony 3 only (routing, etc.) -> maybe directly after a new minor release?

### PR DESCRIPTION
**Pull request by wachterjohannes**
| Q | A
| --- | ---
| Bug fix? | no
| New feature? | no
| BC breaks? | no
| Deprecations? | no
| Fixed tickets | fixes #2106 
| Related issues/PRs | https://github.com/sulu/sulu-standard/pull/795 https://github.com/sulu/sulu-minimal/pull/38
| License | MIT
| Documentation PR | none

#### What's in this PR?

This PR enable sulu to run with symfony 3.0

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -50,19 +51,27 @@
...
+        "doctrine/doctrine-fixtures-bundle": "~2.3",
+        "phpdocumentor/reflection-docblock": "~3.1"
+    },
+    "conflicts": {
+        "symfony/symfony": "2.8.10",
+        "symfony/phpunit-bridge": "2.8.10",
+        "symfony/security-acl": "2.8.10",
+        "symfony/monolog-bundle": "2.8.10"
     },
+    "minimum-stability": "RC",
```
Comment (wachterjohannes) [inline composer.json]: this can be removed when `symfony-cmf/routing-bundle: 2.0` is released.

**Comment by wachterjohannes** [comment]
travis status was no reported but here is the link to the tests https://travis-ci.org/sulu/sulu/builds/194439658

File: src/Sulu/Bundle/ContentBundle/Tests/Functional/Controller/NodeResourcelocatorControllerTest.php [hunk 2]
```diff
@@ -197,8 +197,13 @@ public function testGenerateWithIncompleteParts()
 
     public function testGetAction()
     {
-        // prepare history nodes
-        $newsData = $this->data[1];
+        // prepare history nodes (get to use the newest hash value)
+        $this->client->request(
```
Comment (danrot) [inline src/Sulu/Bundle/ContentBundle/Tests/Functional/Controller/NodeResourcelocatorControllerTest.php]: That shouldn't be necessary after https://github.com/sulu/sulu-document-manager/pull/105

Comment (danrot) [inline src/Sulu/Bundle/ContentBundle/Tests/Functional/Controller/NodeResourcelocatorControllerTest.php]: Actually that is already wrong behavior that should be tested.

File: src/Sulu/Bundle/ContentBundle/Tests/Functional/Controller/NodeResourcelocatorControllerTest.php [hunk 3]
```diff
@@ -221,8 +229,13 @@ public function testGetAction()
 
     public function testDelete()
     {
-        // prepare history nodes
-        $newsData = $this->data[1];
+        // prepare history nodes (get to use the newest hash value)
+        $this->client->request(
```
Comment (danrot) [inline src/Sulu/Bundle/ContentBundle/Tests/Functional/Controller/NodeResourcelocatorControllerTest.php]: Same as before.

**Comment by danrot** [comment]
What changed in the `WebspaceImportTest` file?

**Comment by wachterjohannes** [comment]
this is more developer friendly because the path was "relative" to the root-folder of the repository. if you run phpunit inside a bundle the tests fail.

**Comment by wachterjohannes** [comment]
@danrot we will see if the tests are now running stable without my changes.

**Comment by wachterjohannes** [comment]
i have no idea why travis does not report the state. this is the current test run: https://travis-ci.org/wachterjohannes/sulu

File: composer.json (hunk 1 above)
Comment (danrot) [inline composer.json]: Should we wait until then?

Comment (wachterjohannes) [inline composer.json]: no i would merge it with that. the developer can decide in the real application if he wants that.

File: composer.json [hunk 4]
```diff
@@ -23,16 +23,17 @@
...
         "sulu/document-manager": "0.9.*",
-        "sensio/framework-extra-bundle": "3.0.*",
+        "sensio/framework-extra-bundle": "^3.0",
         "stof/doctrine-extensions-bundle": "^1.2.2",
         "gedmo/doctrine-extensions": "~2.4",
-        "symfony-cmf/routing-bundle": "1.2.*",
-        "symfony/symfony": "~2.8.7, !=2.8.10",
+        "symfony-cmf/routing-bundle": "^1.2 || ^2.0",
+        "symfony/symfony": "^2.8.7 || ^3.0",
+        "symfony/security-acl": "^2.8.7 || ^3.0",
```
Comment (danrot) [inline composer.json]: Why does this have to be added? The only two classes dependent on that are `SymfonyAccessControlManager` and `SymfonyAccessControlVoter`, which both have never really be used, and are marked as deprecated and to be removed in 1.2... I would say it is safe to not add this dependency and remove these two classes.

**Comment by wachterjohannes** [comment]
@danrot fixed and tests seems to work.

**Comment by patkar** [comment]
yayyyyy :tada:

**Pull request merged**